guppylang/prelude/_internal.py

```diff
@@ -271,14 +269,15 @@ def compile(self, args: list[OutPortV]) -> list[OutPortV]:
             float_value(0.0), get_type(self.node), self.dfg.node
         )
         zero = self.graph.add_load_constant(zero_const.out_port(0), self.dfg.node)
-        return Float.__ne__.compile_call(
+        [out] = Float.__ne__.compile_call(
```

Comment:
Why are this and `Float.__truediv__` now returning lists?

Reply:
They also returned a list before, but the issue is that mypy can no longer infer this.

The problem is that `Float` now defines a decorated `__new__` method which throws off mypy, so it can only infer `Float: Any` and thus `Float.__ne__: Any` and `Float.__ne__.compile_call: Any`. By explicitly matching on the list, we can trick mypy into still accepting the code.

Generally, I think that refering back to methods defined on `Float` in this way is more of a hack and not ideal. Once module linking works properly, we can hopefully replace this with a regular Guppy function instead of this hacky `CustomCallCompiler`